internal/services/workloads/workloads_sap_three_tier_virtual_instance_resource_test.go

```diff
@@ -0,0 +1,897 @@
+			),
+		},
+		data.ImportStep(
+			"three_tier_configuration.0.application_server_configuration.0.virtual_machine_configuration.0.os_profile.0.ssh_private_key",
+			"three_tier_configuration.0.central_server_configuration.0.virtual_machine_configuration.0.os_profile.0.ssh_private_key",
+			"three_tier_configuration.0.database_server_configuration.0.virtual_machine_configuration.0.os_profile.0.ssh_private_key",
+		),
+	})
+}
+
+func (r WorkloadsSAPThreeTierVirtualInstanceResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := sapvirtualinstances.ParseSapVirtualInstanceID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	client := clients.Workloads.SAPVirtualInstances
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r WorkloadsSAPThreeTierVirtualInstanceResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "tls_private_key" "test" {
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+data "tls_public_key" "test" {
+  private_key_pem = tls_private_key.test.private_key_pem
+}
+
+data "azurerm_subscription" "current" {}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-sapvis-%d"
+  location = "%s"
+}
+
+resource "azurerm_user_assigned_identity" "test" {
+  name                = "acctest-uai-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_role_assignment" "test" {
+  scope                = data.azurerm_subscription.current.id
+  role_definition_name = "Azure Center for SAP solutions service role"
+  principal_id         = azurerm_user_assigned_identity.test.principal_id
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctest-vnet-%d"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "acctest-subnet-%d"
+  resource_group_name  = azurerm_resource_group.test.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_resource_group" "app" {
+  name     = "acctestRG-sapapp-%d"
+  location = "%s"
+
+  depends_on = [
+    azurerm_subnet.test
+  ]
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r WorkloadsSAPThreeTierVirtualInstanceResource) basic(data acceptance.TestData, sapVISNameSuffix int) string {
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
```

Comment:
Yes. While creating an SAP Three Tier Virtual Instance, the service will provision the extra SAP systems/components in the `app_resource_group_name` that are not defined in the HCL Configuration. At this time, if the `app_resource_group_name` is different from the Resource Group where SAP Three Tier Virtual Instance exists, you can set `prevent_deletion_if_contains_resources` to `false` to delete all resources defined in the HCL Configurations and the resources created in the `app_resource_group_name` with `terraform destroy`. However, if the `app_resource_group_name` is the same with the Resource Group where SAP Three Tier Virtual Instance exists, some resources, such as the subnet defined in the HCL Configuration, cannot be deleted with `terraform destroy` since the resources defined in the HCL Configuration are being referenced by the SAP system/component. In this case, you have to manually delete the SAP system/component before deleting the resources in the HCL Configuration. I also mentioned it in [html md file](https://github.com/hashicorp/terraform-provider-azurerm/pull/24384/files#diff-16c0e0b07b42498840c270ea851bc94122d6bc701715842f9d5e06bca1b3d166R384) in this PR.

Reply:
Thanks for explaining @neil-yechenwei - I did see that in the docs but totally forgot I'd left this comment at that point 😄